src/components/custom-field-input-single-choice/custom-field-input-single-choice.jsx

```diff
@@ -41,35 +62,41 @@ export default function CustomFieldInputSingleChoice(props) {
         placeholder={props.placeholder}
         readOnly={props.readOnly}
         required={props.required}
-        value={props.value}
+        value={value.label}
       />
       { showOptions && (
         <Listbox
           className={styles.dropdown}
           labelledBy={`${props.id}-label`}
+          onSelectionChange={onSelectionChange}
           refs={refs}
         >
-          { props.choices.map(item => <ListOption key={item}>{item}</ListOption>) }
+          { listOptions }
         </Listbox>
       ) }
     </div>
   );
 }
 
+const ChoiceType = PropTypes.shape({
+  id: PropTypes.string,
+  label: PropTypes.string,
+});
+
 CustomFieldInputSingleChoice.propTypes = {
   id: PropTypes.string.isRequired,
-  choices: PropTypes.arrayOf(PropTypes.string),
+  choices: PropTypes.arrayOf(ChoiceType),
   label: PropTypes.string.isRequired,
   placeholder: PropTypes.string,
   readOnly: PropTypes.bool,
   required: PropTypes.bool,
-  value: PropTypes.string,
+  value: ChoiceType,
 };
 
 CustomFieldInputSingleChoice.defaultProps = {
   choices: [],
   placeholder: undefined,
   readOnly: false,
   required: false,
-  value: undefined,
+  value: {},
```

Comment:
Since this is a `ChoiceType` now, should we include an `id` and `label` here in the default? 🤔

Reply:
I thought so, but it was causing spec failures.

Reply:
Would it be better to allow an `undefined` value? I know it makes more code but it would allow us to enforce the `id` and `label` properties of a choice type -- so the question is better phrased: do we think the benefits outweigh the extra code?